api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java

```diff
@@ -189,6 +190,36 @@ public List<Appointment> getAllAppointments(Date forDate) {
         return appointments.stream().filter(appointment -> !isServiceOrServiceTypeVoided(appointment)).collect(Collectors.toList());
     }
 
+    @Transactional
+    @Override
+    public List<Appointment> getAllCameEarlyAppointments(Date forDate) {
+        List<Appointment> appointments = appointmentDao.getAllCameEarlyAppointments(forDate);
+        return appointments.stream().filter(appointment -> !isServiceOrServiceTypeVoided(appointment)).collect(Collectors.toList());
+    }
+
+    @Transactional
+    @Override
+    public List<Appointment> getCompletedAppointments(Date forDate){
+        List<Appointment> appointments = appointmentDao.getCompletedAppointments(forDate);
+        return appointments.stream().filter(appointment -> !isServiceOrServiceTypeVoided(appointment)).collect(Collectors.toList());
+    }
+
+    @Transactional
+    @Override
+    public List<Appointment> getAllAppointments(Date forDate, String status) {
+        List<Appointment> appointments = new ArrayList<Appointment>();
```

Comment:
I don't love the status.trim check on these strings.  Can we look at a better way?